src/crafting.rs

```diff
@@ -341,6 +350,13 @@ pub fn calculate_crafting_profit(
     }
 }
 
+pub struct PurchasedIngredient {
+    pub count: Rational32,
+    pub max_price: i32,
+    pub min_price: i32,
+    pub total_cost: Rational32,
```

Comment:
It might be nice if these had a valid value for vendor items as well. They could either be in an `Option` or just set to the vendor values, since it looks like we want to calculate the total cost later anyway?

Reply:
We do - I chose not to look up the price inside the loop so I'd only have to calculate it once later.